src/plugins/creditor/credNodeStore.test.js

```diff
@@ -0,0 +1,124 @@
+// @flow
+
+import {type CreditorNode} from "./creditorNode";
+import Store, {getDetails, toNodeEntry} from "./credNodeStore";
+import {random as randomUuid} from "../../util/uuid";
+
+describe("plugins/creditor/credNodeStore", () => {
+  const tagId1 = randomUuid();
+  const tagId2 = randomUuid();
+  const baseNode: CreditorNode = {
+    id: randomUuid(),
+    tags: [tagId1, tagId2],
+    title: "Implementing credNodes",
+    description: "creating credNodes for use in the creditor",
+    graphTimestamp: new Date("1/1/2000").getTime(),
+    createdAt: new Date("12/1/2000").getTime(),
+    mint: 0,
+    parent: null,
+  };
+
+  const childNode: CreditorNode = {
+    id: randomUuid(),
+    tags: [tagId2],
+    title: "Creating credNode type",
+    description: "specifying credNodes to test the store",
+    graphTimestamp: new Date("1/1/2000").getTime(),
+    createdAt: new Date("11/1/2000").getTime(),
+    mint: 20,
+    parent: baseNode.id,
+  };
+
+  describe("adding nodes to store", () => {
+    it("requires nodes to have expected members", () => {
+      const store = new Store();
+      const thunk = (badNode: Object) => () => store.set(badNode);
+
+      const badNodes = [
+        [{}, "Creditor Node UUID required"],
+        [{id: randomUuid()}, "`tags` must be an array"],
+        [
+          {id: randomUuid(), tags: []},
+          "`title` and `description` are required in Cred Nodes",
+        ],
+        [
+          {
+            id: randomUuid(),
+            tags: [],
+            title: "bad Node",
+            description: "badDescription",
+            createdAt: 0,
+            graphTimestamp: 0,
+          },
+          "`createdAt` and `graphTimestamp` values must be non-zero",
+        ],
+      ];
+      badNodes.forEach(([node, errorMsg]) =>
+        expect(thunk(node)).toThrowError(errorMsg)
+      );
+    });
+    it("can set and get valid nodes", () => {
+      const store = new Store();
+      const nextStore = store.set(baseNode);
+      expect(nextStore).toEqual(store);
+      const returnedBase = store.get(baseNode.id);
+      expect(returnedBase).toEqual(baseNode);
+    });
+    it("returns null for nonexistent nodes", () => {
+      const store = new Store();
+      const badId = randomUuid();
+      const result = store.get(badId);
+      expect(result).toBeEmpty;
+    });
+  });
+  describe("node interactions", () => {
```

Comment:
we need tests verifying the semantics for replacing a node (same id, new props)
we also need to test how replacement interacts with other API methods. for example, if we have a node with tag Foo, and we replace that node so it now has tag Bar, check that it no longer shows up on a tag search for Foo but it does show up on a tag search for Bar